web/client/epics/catalog.js

```diff
@@ -119,5 +122,80 @@ module.exports = (API) => ({
             .filter((action) => action.control === "metadataexplorer" && action.value)
             .switchMap(() => {
                 return Rx.Observable.of(closeFeatureGrid());
+            }),
+        getMetadataRecordById: (action$, store) =>
+            action$.ofType(GET_METADATA_RECORD_BY_ID)
+            .switchMap(() => {
+                const state = store.getState();
+                const layer = getSelectedLayer(state);
+                return Rx.Observable.fromPromise(
```

Comment:
You should move this code to CSW API, and use a proper require.ensure of utils/ogc/CSW to avoid including ogc-schemas in the main bundle